NEMO/interlocks.py

```diff
@@ -201,6 +201,60 @@ def _send_command(self, interlock: Interlock_model, command_type: Interlock_mode
 			sock.close()
 
 
+class ProXrInterlock(Interlock):
+	"""Very basic support for ProXR relay controllers.
+	Supports on/off/status for all relays in bank 1.
+	See https://ncd.io/proxr-quick-start-guide/ for more about ProXR.
+	"""
+	# supported proxr commands
+	PXR_CMD_B1_OFF = (254, 129, 1)
+	PXR_CMD_B1_ON = (254, 130, 1)
+	PXR_CMD_B1_STAT = (254, 124, 1)
+	# supported proxr responses
+	PXR_RSP_OFF = 0
+	PXR_RSP_ON = 255
+
+	def _send_bytes(self, relay, proxrcmd):
+		"""Returns the response from the relay controller.
+		Argument relay is a connected socket object.
+		Argument proxrcmd (the ProXR command) is an iterable of 8-bit integers.
+		"""
+		# make sure that all bytes get sent
+		bytes_sent = 0
+		while bytes_sent < len(proxrcmd):
+			bytes_sent += relay.send(bytes(proxrcmd[bytes_sent:]))
+		# relay responses can include erroneous data
+		# only the last byte of the response is important
+		resp = relay.recv(64)[-1]
+		return resp
+
+	def _send_command(self, interlock: Interlock_model, command_type: Interlock_model.State) -> Interlock_model.State:
+		"""Returns current NEMO locked/unlocked state.
+		Toggles the relay on/off.
+		"""
+		# get the proxr state before operation
+		# toggle the relay on/off
+		# get the proxr state after operation
+		# return the nemo state
+		after = None
+		try:
+			with socket.create_connection((interlock.card.server, interlock.card.port), 10) as relay:
+				before = self._send_bytes(relay, self.PXR_CMD_B1_STAT)
+				if before == self.PXR_RSP_OFF:
+					self._send_bytes(relay, self.PXR_CMD_B1_ON)
+				elif before == self.PXR_RSP_ON:
+					self._send_bytes(relay, self.PXR_CMD_B1_OFF)
+				after = self._send_bytes(relay, self.PXR_CMD_B1_STAT)
```

Comment:
so this really doesn't take into account the command sent by NEMO. it just toggles the state regardless.
NEMO has functions to lock/unlock interlocks, and the expected behavior is that the result state is locked/unlocked.
It would make more sense to simply send the on or off command based on `command_type` like it's done in the other interlock implementations. 

If you want it to be safe, you can stat the interlock, and if it's already locked when lock command is sent, do nothing (I would put a warning log message still). same would go for unlocking.
does that make sense?

Reply:
Makes sense. Changes are done and seem to work well. Pushed to my master.